use_cluster.py

```diff
@@ -366,4 +355,411 @@ def return_cluster_dict(q_cluster, q_ids_list):
+    scores_1gram = calculate_label_score(cluster_objs, vecs, global_keywords, lemmatized_qs_list)
+    
+    
+    cluster_objs, lemmatized_qs_list = generate_modified_qs_data(labelling_corpus.documents, clusters, q_ids_list,2)
+    global_keywords = generate_global_keywords(labelling_corpus.documents, 2)
+    vecs = embed(list(global_keywords))
+    scores_2gram = calculate_label_score(cluster_objs, vecs, global_keywords, lemmatized_qs_list)
+    
+    combined_scores = []
+    for score_1gram, score_2gram in zip(scores_1gram, scores_2gram):
+        score_1gram.extend(score_2gram)
+        score_1gram.sort(key = lambda x: x[1],reverse=True)
+        combined_scores.append(score_1gram)
+
+    final_labels = return_best_label_combination(combined_scores)
+    return final_labels
+
+"""
+    >Used to Check Labels Generated from Label Scores
+    >Use this code snippet before returning the labels
+
+    for label, clus_score in zip(final_labels, combined_scores):
+        print(clus_score[:5])
+        print(label)
+        print("-------------")
+"""    
+    
+
+"""
+> The objective of this function is to return whether a label is 1 gram or 2 gram:
+1. In a 1 gram, it's one word, with no spaces
+2. In a 2 gram, there are two words, with a space in between
+4. So this function checks for the space and returns the gram type
+5. O/P is 1 for 1 gram & 2 for 2gram
+:The inputs required are just the label, which is a string  
+"""
+
+def get_label_type(label):
+    if ' ' in label:
+        return 2 #2gram
+    else:
+        return 1 #1gram
+
+"""
+> The objective of this function is to decide the final labels for clusters:
+1. We will only consider 2 Keywords (1 or 2 grams)
+2. The algorithm to select these 2 Keywords is basically a set of if-else conditions 
+3. It is broken down into 4 cases, based on if the top 2 labels by score are a particular combination of 1 or 2 grams 
+4. For each case, certain conditions are checked, to return the best set of labels for that cluster
+5. A diagram for this algorithm, created by Araz is found here: https://drive.google.com/file/d/1v_lFS_jUgqzPfRGNPpCcuZwHtdVF1BuO/view?usp=sharing
+:The inputs required are the label scores cluster wise 
+"""
+
+
+def return_best_label_combination(label_scores):
+    final_labels = []
+    for cluster_labels in label_scores:
+        label_1 = cluster_labels[0][0]
+        label_2 = cluster_labels[1][0]
+        label_3 = cluster_labels[2][0]
+        label_4 = cluster_labels[3][0]
+
+        label_1_type = get_label_type(label_1)
+        label_2_type = get_label_type(label_2)       
+        label_3_type = get_label_type(label_3)
+        label_4_type = get_label_type(label_4)       
+
+        
+        #CASE I label_1 1 gram label_2 2 gram
+
+        if(label_1_type==1 and label_2_type==2):
+            if(label_1 not in label_2):
+                selected = [label_1, label_2]
+                final_labels.append(selected)
+            else:
```

Comment:
I think this would read a bit more clearly if you used "elif (label_3_type === 2)" instead of nesting more deeply after each else. And the same for the subsequent if/else blocks.

Reply:
Yes, agreed! Have made this change wherever applicable :D